Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM centos:centos7
+LABEL maintainer="Nik Hodgkinson <11xor6@gmail.com>"
+
+ENV JAVA_HOME /usr/lib/jvm/java-1.11.0
+RUN yum -y install epel-release && \
+    yum -y update && \
+    yum -y install java-1.11.0-openjdk-devel python-pip && \
+    yum clean all && \
+    rm -rf /var/cache/yum && \
+    pip install --upgrade pip && \
+    pip install dumb-init
+
+ARG PRESTO_VERSION
+RUN mkdir -p /data/presto && \
+    mkdir -p /usr/lib/presto/etc/catalog && \
+    cd /tmp && \
+    curl http://central.maven.org/maven2/io/prestosql/presto-server/${PRESTO_VERSION}/presto-server-${PRESTO_VERSION}.tar.gz | tar xz && \
+    mv presto-server-${PRESTO_VERSION}/* /usr/lib/presto && \
+    curl http://central.maven.org/maven2/io/prestosql/presto-cli/${PRESTO_VERSION}/presto-cli-${PRESTO_VERSION}-executable.jar -o /usr/bin/presto && \
+    chmod 755 /usr/bin/presto
+
+EXPOSE 8080
+VOLUME /usr/lib/presto/etc /usr/lib/presto/etc/catalog
```

Comment:
why do we need `/usr/lib/presto/etc/catalog` volume if we already has `/usr/lib/presto/etc` that encloses this location?

Reply:
+1. Wouldn't `VOLUME` be confusing for users? Specifically:
1. additional step to remove such volume after container is removed
2. Is volume preserved between container versions?

I would prefer image customization via additional bootstrapping (e.g: via explicitly mounting catalogs).

Reply:
The `VOLUME` command in a `Dockerfile` is purely to specify the mount points that exist. No actual volumes are created. It is optional to actually mount something to that location at runtime. 

Additionally, I listed both `/usr/lib/presto/etc` and `/usr/lib/presto/etc/catalog` as those could possibly be two separate mount points or alternatively the user can mount a single config volume at `/usr/lib/presto/etc`.

Reply:
From https://docs.docker.com/engine/reference/builder/#volume

> This Dockerfile results in an image that causes docker run to create a new mount point at /myvol and copy the greeting file into the newly created volume.

This sounds like `VOLUME` is not "purely documentational" .

Reply:
After doing quite a bit of research on Docker volumes, I'm inclined to agree here. I didn't realize that volumes have a separate lifecycle and that doesn't seem appropriate for configuration. I do want to raise the issue of `/data/presto` being a volume however, it debatably is a proper use.

Reply:
Presto does utilize persistent storage. What's the purpose?

Reply:
It may be an isolated use case and I'm happy to remove it as a volume, however it seems to me that if a user were using a persistent storage method like raptor they may want that data to persist between runs of their container. Otherwise all data would be lost in a container restart or shutdown. However it's also quite possible for someone to take our image and add a volume to it as a new image built upon this one. In the case of utilizing this image with Kubernetes volumes are completely unnecessary and thus this only really applies to anyone running the container in some local or Docker Swarm environment.

Reply:
@11xor6 lack of `VOLUME` declaration does not prevent anyone from attaching a volume at run-time, even without having downstream docker image.
If this isn't necessary for your use-case, let leave it out.

Reply:
This isn't necessary for the Kubernetes operator, I'll remove.